src/dev/i18n/extract_default_translations.js

```diff
@@ -38,15 +40,56 @@ function addMessageToMap(targetMap, key, value) {
   targetMap.set(key, value);
 }
 
-export async function extractDefaultTranslations(inputPath) {
+async function getPluginsPaths(inputPaths) {
+  const availablePaths = Object.values(config.paths);
+  const pathsForExtraction = [];
+
+  for (const inputPath of inputPaths) {
+    const normalizedPath = normalize(path.relative('.', inputPath));
+
+    if (normalizedPath) {
+      pathsForExtraction.push(
+        ...availablePaths.filter(
+          ePath => ePath.startsWith(`${normalizedPath}/`) || ePath === normalizedPath
+        )
+      );
+    } else {
+      pathsForExtraction.push(...availablePaths);
+    }
+
+    if (
+      availablePaths.some(
+        ePath => normalizedPath.startsWith(`${ePath}/`) || ePath === normalizedPath
+      )
+    ) {
+      pathsForExtraction.push(normalizedPath);
+    }
+  }
+
+  return pathsForExtraction;
+}
+
+export function validateMessageNamespace(id, filePath) {
+  const normalizedPath = normalize(path.relative('.', filePath));
+
+  const [expectedNamespace] = Object.entries(config.paths).find(([, pluginPath]) =>
+    normalizedPath.startsWith(`${pluginPath}/`)
+  );
+
+  if (!id.startsWith(`${expectedNamespace}.`)) {
+    throw new Error(`Message id has a wrong namespace ("${id}").`);
```

Comment:
__nit:__ please say something like `expected this .... got this .... See i18nrc.json for the list of supported namespaces`...